src/bootstrap/src/core/config/config.rs

```diff
@@ -2847,6 +2834,42 @@ impl Config {
 
         Some(commit.to_string())
     }
+
+    /// Check for changes in specified directories since a given commit.
+    /// Returns Some(true) if changes exist, Some(false) if no changes, None if check should be ignored.
+    pub fn check_for_changes(
+        &self,
+        dirs: &[PathBuf],
+        commit: &str,
+        option_name: &str,
+        if_unchanged: bool,
+    ) -> Option<bool> {
```

Comment:
I think this should have only one purpose: check whether there are changes or not. There is no need to add `if_unchanged` argument and an optional result type.

Reply:
yes. done.